mobile/cmp/navigator/NavigatorModel.ts

```diff
@@ -131,8 +141,89 @@ export class NavigatorModel extends HoistModel {
     //--------------------
     // Implementation
     //--------------------
-    private onRouteChange(init = null) {
-        if (!this._navigator || !XH.routerState) return;
+    /** @internal */
+    setSwiper(swiper: Swiper) {
+        if (this._swiper) return;
+        this._swiper = swiper;
+
+        swiper.on('transitionEnd', () => this.onPageChange());
+
+        // Ensure Swiper's touch move is initially disabled, and capture
+        // the initial touch position. This is required to allow touch move
+        // to propagate to scrollable elements within the page.
+        swiper.on('touchStart', (s, event: PointerEvent) => {
+            swiper.allowTouchMove = false;
+            this._touchStartX = event.pageX;
+        });
+
+        // Add our own "touchmove" handler to the swiper, allowing us to toggle
+        // the built-in touch detection based on the presence of scrollable elements.
+        swiper.el.addEventListener('touchmove', (event: TouchEvent) => {
+            const touch = event.touches[0],
+                distance = touch.clientX - this._touchStartX,
+                direction = distance > 0 ? 'right' : 'left';
+
+            const scrollableParent = findScrollableParent(event, 'horizontal');
+            if (scrollableParent) {
```

Comment:
do I dare ask about nested scrollable parents?  Seems uncommon, but....

Reply:
You mean a scrollable within a scrollable within a swiper... 🤯 

Currently, the one you are touching will scroll until the left edge and then the swiper will come into play, skipping the middle scrollable.

Do we have a specific need for it? If not I'd vote with live with it until it comes up, especially given that we should try to avoid horizontal scrolling on mobile generally outside of this context